'tmppath may be used uninitialized' warning during the building of rpm

Hi @brauner 

STR:
-clone master branch and try to build the RPM

Actual Result:
```
storage/btrfs.c: In function 'btrfs_recursive_destroy':
storage/btrfs.c:847:10: error: 'tmppath' may be used uninitialized in this function [-Werror=maybe-uninitialized]
      free(tmppath);
          ^
cc1: all warnings being treated as errors
make[4]: *** [storage/liblxc_la-btrfs.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../
make[3]: *** [all-recursive] Error 1
....
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.q9KxAO (%build)
make: *** [rpm] Error 1
```

As far as I can see you added these changes in https://github.com/lxc/lxc/pull/2900 Could you please fix this warning?

@brauner 

I just tried to build the RPM from master branch and the error is till there.
```
07:57:03 root#configurator:~/lxc
> git log --name-status HEAD^..HEAD
commit 9810d1916eacea64c838681d3045cac60e7f00c2 (HEAD -> master, origin/master, origin/HEAD)
Merge: 74f628ac 13f76743
Author: Christian Brauner <christian@brauner.io>
Date:   Sat Apr 6 15:14:31 2019 +0200

    Merge pull request #2928 from yosukesan/api_doc_generation_fails_under_out_of_source_build#2927
```

Here is a part from 'make rpm' output:
```
test -f 'storage/storage_utils.c' || echo './'`storage/storage_utils.c
storage/btrfs.c: In function 'btrfs_recursive_destroy':
storage/btrfs.c:847:10: error: 'tmppath' may be used uninitialized in this function [-Werror=maybe-uninitialized]
      free(tmppath);
```

--- a/src/lxc/storage/btrfs.c
+++ b/src/lxc/storage/btrfs.c
@@ -41,6 +41,7 @@
 #include "btrfs.h"
 #include "config.h"
 #include "log.h"
+#include "memory_utils.h"
 #include "rsync.h"
 #include "storage.h"
 #include "utils.h"
@@ -763,7 +764,6 @@ static int btrfs_recursive_destroy(const char *path)
 	int ret, e, i;
 	unsigned long off = 0;
 	u16 name_len;
-	char *tmppath;
 	u64 dir_id;
 
 	fd = open(path, O_RDONLY);
@@ -834,7 +834,8 @@ static int btrfs_recursive_destroy(const char *path)
 			 * name of the child subvol in question.
 			 */
 			if (sh.objectid != root_id && sh.type == BTRFS_ROOT_BACKREF_KEY) {
-				char *name, *tmp;
+				__do_free char *name = NULL, *tmppath = NULL;
+				char *tmp;
 
 				ref = (struct btrfs_root_ref *)(args.buf + off);
 				name_len = btrfs_stack_root_ref_name_len(ref);
@@ -844,7 +845,6 @@ static int btrfs_recursive_destroy(const char *path)
 				if (!name) {
 					ERROR("Out of memory");
 					free_btrfs_tree(tree);
-					free(tmppath);
 					close(fd);
 				}
 
@@ -859,14 +859,9 @@ static int btrfs_recursive_destroy(const char *path)
 							name_len, tmppath)) {
 					ERROR("Out of memory");
 					free_btrfs_tree(tree);
-					free(name);
-					free(tmppath);
 					close(fd);
 					return -1;
 				}
-
-				free(tmppath);
-				free(name);
 			}
 
 			off += sh.len;

